stock_mts_mto_rule/models/stock_rule.py

```diff
@@ -26,14 +35,18 @@ def _check_mts_mto_rule(self):
                     ) % (rule.name,)
                     raise ValidationError(msg)
 
+    @api.model
+    def _get_available_quantity_mts_mto_rule(self, product):
+        return product.virtual_available
+
     def get_mto_qty_to_order(self, product, product_qty, product_uom, values):
         self.ensure_one()
         precision = self.env["decimal.precision"].precision_get(
             "Product Unit of Measure"
         )
         src_location_id = self.mts_rule_id.location_src_id.id
         product_location = product.with_context(location=src_location_id)
-        virtual_available = product_location.virtual_available
+        virtual_available = self._get_available_quantity_mts_mto_rule(product_location)
```

Comment:
Could we simplify this computation?
And adding the check for the mts_mto_split_quantity_rule here instead of in _run_split_procurement?

```suggestion
        virtual_available = self._get_available_quantity_mts_mto_rule(product_location)
        qty_available = product.uom_id._compute_quantity(virtual_available, product_uom)
        qty_mto = product_qty - qty_available
        if float_compare(qty_mto, 0, precision_digits=precision) <= 0:
            return 0
        if rule.mts_mto_split_quantity_rule == "full":
            return product_qty
        return qty_mto
```

Reply:
I am not sure of this.
We don't have access to `rule` in this function, and I don't like changing the signature of an existing function.
And it also makes the diff bigger.

Reply:
we have access to the rule it is self.

Reply:
yes it makes the diff bigger, but i think it makes it much more understandable.
The method _run_split_procurement would not change. 

Because we only need to change the returned qty.

Reply:
> we have access to the rule it is self.

oh :blush:  you are right

Reply:
@TDu do you want to simplify the computation? :-)

Reply:
ping @TDu ?

Reply:
Thanks for your review. But I would like to keep the computation as it is, now.